### ISSUE TITLE
User can define their preferred language in OJS

### ISSUE THREAD

**Issue by pilasou** [pkp/pkp-lib#7293]
**Describe the problem you would like to solve**
Users cannot define a preferred language for the OJS UI.

We are using OJS 3.3.0.6 in a multijournal environment. Many of our journal are bilingual (french/english), with french being the default language configured. 

The way OJS manage the UI language causes some problems to english speaking authors and reviewers as they do not have the opportunity to setup a language preference in OJS.

**Describe the solution you'd like**
Use Case: As a user, I want to be able to setup a prefered language so that all informations remain in my own language.

1.	Journals still have control on the languages available in their interface (define which langage to display and which is the default language).
2.	Default user's language is the one define by the journal as the default language.
3.	Journals can define the prefered language for each user. 
4.	If the journal offers more than one language, when registering, a user can define his/hers prefered language among the languages available.
5.  A user can change his/her prefered language at any time in their profil information page. 
6.  The value of the prefered language takes precedence over the default language define by the journal.
7.  The UI can at any time be changed to any language configured by the journal, no matter the preferred language setted up by the user.

There is a language field associated with each OJS users account (Working Languages) that may be used as the preferred language as describe above.

**Who is asking for this feature?**
Problem were reported to us by journal editors. Authors and Reviewers complain about this problems.

**Comment by asmecher** [comment]
There is a "working languages" user setting available on the user's profile; is the issue that this setting isn't widely used? See related issues:
- https://github.com/pkp/pkp-lib/issues/6221
- https://github.com/pkp/pkp-lib/issues/3525

**Comment by pilasou** [comment]
Hi @asmecher and @NateWr  Thanks, the problem is more that working language has no effect whatsoever on the UI language.  When you change a UI language in OJS, either being connected/registered or not, this change will end with your session. If you disconnect, close your browser and come back and reconnect, you will have to change the UI language again.

**Comment by NateWr** [comment]
Ah, I see what you're saying. If the language selection persisted across browser sessions, would that resolve the issue you describe?

(@asmecher I'm wondering if it makes sense to use a persistent cookie for language selection? I think this can be considered a core use and therefore we don't need a GDPR cookie banner.)

**Comment by pilasou** [comment]
Yes, the idea is that the language selected by the user remains across sessions. With regards to GDPR, my understanding is that you do not need consent for "strictly necessary cookie", meaning cookies needed for basics fonctions of the website. Can language cookie be consider a strictly necessary cookie for OJS? (I love this idea!) :)

**Comment by NateWr** [comment]
Thanks @pilasou. I've edited the issue title and description to reflect this solution.
